### PR TITLE
[terra-application-link] Add theme context

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Added theme context to terra application link
### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-application-links/src/tabs/ApplicationTabs.jsx [hunk 1]
```diff
@@ -176,19 +178,26 @@ class ApplicationTabs extends React.Component {
...
+    const tabClass = classNames(cx(
+      'tabs-container',
+      { 'is-calculating': this.isCalculating },
+      alignment,
+      theme.className,
+    ));
 
     return (
-      <div {...customProps} className={cx(['tabs-wrapper'])}>
+      <div {...customProps} className={classNames(cx('tabs-wrapper', theme.className))}>
```
Comment (StephenEsser) [inline packages/terra-application-links/src/tabs/ApplicationTabs.jsx:190]: Something isn't quite right here. The theme class name is being applied to this div as well as the div below

File: packages/terra-application-links/src/tabs/ApplicationTabs.jsx [hunk 2]
```diff
@@ -176,19 +178,26 @@ class ApplicationTabs extends React.Component {
         collapsedTabs.push(<CollapsedTab {...tabProps} />);
       }
     });
+    const theme = this.context;
+    const tabClass = classNames(cx(
+      'tabs-container',
+      { 'is-calculating': this.isCalculating },
+      alignment,
+      theme.className,
+    ));
```
Comment (StephenEsser) [inline packages/terra-application-links/src/tabs/ApplicationTabs.jsx:187]: This class list is being applied to an inner div tabs wrapper, adding the theme class name here probably isn't necessary. 

Wrapping `classNames` and `cx` together is used when applying customProps.classNames to the class list without binding it to the stylesheet. If an unbound class isn't being used you don't need to wrap them together, Only `cx` is necessary.

```js
classNames([
  cx('bound-class'), 
  customProps.className // Add the custom class to the classList without binding it to the stylesheet
])
```

File: packages/terra-application-links/src/tabs/ApplicationTabs.jsx [hunk 3]
```diff
@@ -176,19 +178,26 @@ class ApplicationTabs extends React.Component {
...
+          className={tabClass}
           role="tablist"
           ref={this.setContainerNode}
         >
           {visibleTabs}
           <TabMenu location={location} isHidden={this.menuHidden} hasIcons={hasIcons}>
             {collapsedTabs}
           </TabMenu>
-          <div className={cx(['divider-after-last-tab'])} />
+          <div className={classNames(cx('divider-after-last-tab', theme.className))} />
```
Comment (StephenEsser) [inline packages/terra-application-links/src/tabs/ApplicationTabs.jsx:200]: The `theme.className` binding only needs to be spread on the root element

**Comment by saket2403** [comment]
resolved comments dc9bae1

File: packages/terra-application-links/src/tabs/ApplicationTabs.jsx [hunk 4]
```diff
@@ -176,19 +176,24 @@ class ApplicationTabs extends React.Component {
...
     });
+    const tabClass = cx(
+      'tabs-container',
+      { 'is-calculating': this.isCalculating },
+      alignment,
+    );
 
     return (
-      <div {...customProps} className={cx(['tabs-wrapper'])}>
+      <div {...customProps} className={cx('tabs-wrapper')}>
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/tabs/ApplicationTabs.jsx:186]: This should have applied  the theme context class name here

Comment (saket2403) [reply packages/terra-application-links/src/tabs/ApplicationTabs.jsx]: Added 32feabb

File: packages/terra-application-links/src/tabs/ApplicationTabs.module.scss [hunk 5]
```diff
@@ -1,3 +1,8 @@
+
+// Themes
+@import './clinical-lowlight-theme/ApplicationLinks.module';
+@import './orion-fusion-theme/ApplicationLinks.module';
+
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/tabs/ApplicationTabs.module.scss:5]: These should be importing from 
```suggestion
@import './clinical-lowlight-theme/ApplicationTabs.module';
@import './orion-fusion-theme/ApplicationTabs.module';

```

Comment (saket2403) [reply packages/terra-application-links/src/tabs/ApplicationTabs.module.scss]: Done 1fa7f41

File: packages/terra-application-links/src/tabs/clinical-lowlight-theme/ApplicationTabs.module.scss [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    --terra-application-links-tabs-tab-selected-divider-before-background-image: none;
+    --terra-application-links-tabs-tab-selected-divider-before-border-left-color: #022b46;
+    --terra-application-links-tabs-tab-selected-divider-before-border-right-color: #0f699f;
+    --terra-application-links-tabs-tab-selected-font-weight: 500;
+    --terra-application-links-tabs-tab-selected-text-shadow: none;
+    --terra-application-links-tabs-tab-selected-background-size: 13.215rem 6px;
+
+    --terra-application-links-tabs-wrapper-padding-left: 1.42857rem;
+    --terra-application-links-tabs-wrapper-padding-right: 1.42857rem;
+
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/tabs/clinical-lowlight-theme/ApplicationTabs.module.scss:78]: nit: extra line

File: packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    --terra-application-links-tabs-tab-selected-divider-before-background-image: none;
+    --terra-application-links-tabs-tab-selected-divider-before-border-left-color: rgba(255, 255, 255, 0.17);
+    --terra-application-links-tabs-tab-selected-divider-before-border-right-color: rgba(0, 0, 0, 0.17);
+    --terra-application-links-tabs-tab-selected-font-weight: 500;
+    --terra-application-links-tabs-tab-selected-text-shadow: none;
+    --terra-application-links-tabs-tab-selected-background-size: 13.215rem 6px;
+
+    --terra-application-links-tabs-wrapper-padding-left: 1.42857rem;
+    --terra-application-links-tabs-wrapper-padding-right: 1.42857rem;
+
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss:78]: nit: extra line

File: packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss [hunk 8]
```diff
@@ -0,0 +1,80 @@
+:local {
+  .orion-fusion-theme {
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss:2]: In https://github.com/cerner/terra-framework/blob/master/packages/terra-application-links/src/tabs/ApplicationTabs.module.scss there are 68 theme vars, but here there are 65. Are there any missing?

Comment (saket2403) [reply packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss:2]: Actually a couple of variables are used twice, once with calc function like [here](https://github.com/cerner/terra-framework/blob/b123efc4132de8e1b2235bd3e6a0e7dbbf4ae032/packages/terra-application-links/src/tabs/ApplicationTabs.module.scss#L22). No variables are missing

File: packages/terra-application-links/src/tabs/clinical-lowlight-theme/ApplicationTabs.module.scss (hunk 6 above)
Comment (saket2403) [reply packages/terra-application-links/src/tabs/clinical-lowlight-theme/ApplicationTabs.module.scss]: Removed

File: packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss (hunk 7 above)
Comment (saket2403) [reply packages/terra-application-links/src/tabs/orion-fusion-theme/ApplicationTabs.module.scss]: Removed

**Pull request merged**